src/proxies/sitesapi.js

```diff
@@ -0,0 +1,52 @@
+const platformClient = require('purecloud-platform-client-v2');
+
+const sitesMap = new Map();
+
+/*
+    The getSiteByLogicalName() will look up a site based on its logical name from GenesysCloud from Genesys Cloud 
+*/
+const getSiteByLogicalName = async (logicalName) => {
+  let opts = {
+    name: logicalName,
+  };
+
+  let apiInstance = new platformClient.TelephonyProvidersEdgeApi();
+
+  try {
+    const results = await apiInstance.getTelephonyProvidersEdgesSites(opts);
+
+    const site = {
+      id: results.entities[0].id,
+      name: results.entities[0].name,
+      primarySites: results.entities[0].primarySites,
+    };
+
+    if (results != null) {
+      sitesMap[site.name] = site;
+      return {...site};
+    }
+
+    return null;
+  } catch (err) {
+    console.log(
+      `Error while retrieving site with name: ${logicalName}: ${JSON.stringify(
+        err,
+        null,
+        '\t'
+      )}`
+    );
+    return null;
+  }
+};
+
+const getSiteByName = async (siteName) => {
+  const results = sitesMap[siteName];
```

Comment:
I haven't commented on this anywhere else yet, but results is not a normal name in JavaScript. Its too arbitrary. Typically you would say `const site = sitesMap[siteName]`

Also, a more succinct way to write this would be.

```suggestion
  return sitesMap[siteName] || await getSiteByLogicalName(siteName);
```